src/AutoMerge/new-version.jl

```diff
@@ -46,57 +47,49 @@ function pull_request_build(data::GitHubAutoMergeData, ::NewVersion)::Nothing
         this_pr_can_use_special_jll_exceptions = false
     end
 
-    guidelines = Guideline[]
-    if !this_is_jll_package && pr_author_login ∉ data.authorized_authors
-        push!(guidelines,
-              guideline_jll_only_authorization) # 0
-    end
-
-    push!(guidelines,
-          guideline_pr_only_changes_allowed_files) # 1
+    jll_only_authorization = (!this_is_jll_package
+                              && pr_author_login ∉ data.authorized_authors)
 
-    if !this_pr_can_use_special_jll_exceptions
-        push!(guidelines,
-              guideline_sequential_version_number) # 2
-    end
+    # Each element is a tuple of a guideline and whether it's
+    # applicable. Instead of a guideline there can be the symbol
+    # `:update_status` in which case the PR status will be updated
+    # with the results so far before continuing to the following
+    # guidelines.
+    guidelines =
+        [(guideline_jll_only_authorization, jll_only_authorization), #0
+         (guideline_pr_only_changes_allowed_files, true), # 1
+         (guideline_sequential_version_number,
+          !this_pr_can_use_special_jll_exceptions), # 2
+         (guideline_compat_for_all_deps, true), # 3
+         (guideline_patch_release_does_not_narrow_julia_compat,
+          !this_pr_can_use_special_jll_exceptions), # 4
+         (guideline_allowed_jll_nonrecursive_dependencies,
+          this_is_jll_package), # 5
+         (:update_status, true),
+         (guideline_version_can_be_pkg_added, true), # 6
+         (guideline_version_can_be_imported, true)] # 7
 
-    push!(guidelines,
-          guideline_compat_for_all_deps) # 3
-
-    if !this_pr_can_use_special_jll_exceptions
-        push!(guidelines,
-              guideline_patch_release_does_not_narrow_julia_compat) # 4
-    end
+    checked_guidelines = Guideline[]
 
-    push!(guidelines,
-          guideline_allowed_jll_nonrecursive_dependencies) # 5
```

Comment:
Note: This guideline should only be applied to jll packages. It's a bug from #318 which is fixed in this PR. It's a bit worrying that the tests didn't catch it.

Reply:
good catch! looks like we did it right in `new-package.jl` but not here. As you mentioned, I think we are probably not covering that case in the integration tests. If you are up for it, you could look at adding one; the way it works is there's a list of folders that work like the "feature branch" a registration PR might make, and a list of "trunk branch" folders that correspond to possible states of the registry. Then we loop through and check that various pairs of feature-branch/trunk-branch pass or fail as appropriate.